src/controllers/blocks/BlocksController.ts

```diff
@@ -169,4 +166,77 @@ export default class BlocksController extends AbstractController<BlocksService>
 			await this.service.fetchBlock(hash, options)
 		);
 	};
+
+	/**
+	 * Return the Header of a block when a block number is queried
+	 *
+	 * @param req Express Request
+	 * @param res Express Response
+	 */
+	private getBlockByIdSummary: RequestHandler<INumberParam> = async (
+		{ params: { number } },
+		res
+	): Promise<void> => {
+		const hash = await this.getHashForBlock(number);
+
+		BlocksController.sanitizedSend(
+			res,
+			await this.service.fetchBlockSummary(hash)
+		);
+	};
+
+	/**
+	 * Return the header of the latest block
+	 *
+	 * @param req Express Request
+	 * @param res Express Response
+	 */
+	private getLatestBlockSummary: RequestHandler = async (
+		{ query: { finalized } },
+		res
+	): Promise<void> => {
+		const paramFinalized = finalized === 'true' ? true : false;
+
+		const hash = !paramFinalized
+			? await this.api.rpc.chain.getFinalizedHead()
```

Comment:
This query gets the finalized head, but the way I read it we do this query when we _don't_ want the finalized head